bundles/core/org.eclipse.smarthome.core.binding.xml.test/src/test/java/org/eclipse/smarthome/core/binding/xml/test/BindingInfoI18nTest.java

```diff
@@ -0,0 +1,148 @@
+        assertThat(bindingInfo.getDescription(), is(
+                "Das Yahoo Wetter Binding stellt verschiedene Wetterdaten wie die Temperatur, die Luftfeuchtigkeit und den Luftdruck für konfigurierbare Orte vom yahoo Wetterdienst bereit"));
+    }
+
+    @Test
+    public void assertBindingInfosWereLocalizedInDutch() throws Exception {
+        BundleContext bundleContext = getBundleContext();
+        int initialNumberOfBindingInfos = bindingInfoRegistry.getBindingInfos().size();
+
+        // install test bundle
+        Bundle bundle = SyntheticBundleInstaller.install(bundleContext, TEST_BUNDLE_NAME);
+        assertThat(bundle, is(notNullValue()));
+
+        Set<BindingInfo> bindingInfos = bindingInfoRegistry.getBindingInfos(new Locale("nl"));
+        assertThat(bindingInfos.size(), is(initialNumberOfBindingInfos + 1));
+        BindingInfo bindingInfo = bindingInfos.iterator().next();
+
+        assertThat(bindingInfo, is(notNullValue()));
+        assertThat(bindingInfo.getName(), is("Yahoo Weer Binding"));
+        assertThat(bindingInfo.getDescription(), is(
+                "De Yahoo Weer Binding biedt verschillende meteorologische gegevens zoals temperatuur, vochtigheid en luchtdruk voor configureerbare locaties uit yahoo weerdienst klaar"));
+    }
+
+    @Test
+    public void assertUsingOriginalBindingInfosIfProvidedLocaleIsNotSupported() throws Exception {
+        BundleContext bundleContext = getBundleContext();
+        int initialNumberOfBindingInfos = bindingInfoRegistry.getBindingInfos().size();
+
+        // install test bundle
+        Bundle bundle = SyntheticBundleInstaller.install(bundleContext, TEST_BUNDLE_NAME);
+        assertThat(bundle, is(notNullValue()));
+
+        Set<BindingInfo> bindingInfos = bindingInfoRegistry.getBindingInfos(Locale.FRENCH);
+        assertThat(bindingInfos.size(), is(initialNumberOfBindingInfos + 1));
+        BindingInfo bindingInfo = bindingInfos.iterator().next();
+
+        assertThat(bindingInfo, is(notNullValue()));
+        assertThat(bindingInfo.getName(), is("YahooWeather Binding"));
+        assertThat(bindingInfo.getDescription(), is(
+                "The Yahoo Weather Binding requests the Yahoo Weather Service to show the current temperature, humidity and pressure."));
+    }
+
+    @Test
+    public void assertUsingDefaultLocale() throws Exception {
+        // Set german locale
+        ConfigurationAdmin configAdmin = getService(ConfigurationAdmin.class);
+        Configuration config = configAdmin.getConfiguration("org.eclipse.smarthome.core.i18nprovider", null);
+        Dictionary<String, String> localeCfg = new Hashtable<>();
+        localeCfg.put("language", "de");
+        localeCfg.put("country", "DE");
+        config.update(localeCfg);
+
+        // before running the test with a default locale make sure the locale has been set
+        LocaleProvider localeProvider = getService(LocaleProvider.class);
+        waitForAssert(new Closure<Object>(null) {
```

Comment:
Looks like you currently can only use Groovy Closures with waitForAssert in the OSGiTest. There are no methods with Suppliers like in the JavaTest.